test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/FreezeUnfreezeTokenPrecompileSuite.java

```diff
@@ -95,7 +93,136 @@ public List<HapiApiSpec> getSpecsInSuite() {
+                                                        .via("FreezeTx"))))
+                .then(
+                        childRecordsCheck(
+                                "UnfreezeTx",
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_ID)),
+                        childRecordsCheck(
+                                "FreezeTx",
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_ID)));
+    }
+
+    private HapiApiSpec noKeyReverts() {
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
+        final AtomicReference<TokenID> withoutKeyID = new AtomicReference<>();
+
+        return defaultHapiSpec("noKeyReverts")
+                .given(
+                        newKeyNamed(FREEZE_KEY),
+                        newKeyNamed(MULTI_KEY),
+                        cryptoCreate(ACCOUNT)
+                                .balance(100 * ONE_HBAR)
+                                .exposingCreatedIdTo(accountID::set),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(TOKEN_WITHOUT_KEY)
+                                .exposingCreatedIdTo(id -> withoutKeyID.set(asToken(id))),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .freezeKey(FREEZE_KEY)
+                                .initialSupply(1_000)
+                                .exposingCreatedIdTo(id -> vanillaTokenID.set(asToken(id))),
+                        uploadInitCode(FREEZE_CONTRACT),
+                        contractCreate(FREEZE_CONTRACT),
+                        tokenAssociate(ACCOUNT, VANILLA_TOKEN))
+                .when(
+                        withOpContext(
+                                (spec, opLog) ->
+                                        allRunFor(
+                                                spec,
+                                                contractCall(
+                                                                FREEZE_CONTRACT,
+                                                                TOKEN_UNFREEZE_FUNC,
+                                                                asAddress(withoutKeyID.get()),
+                                                                asAddress(accountID.get()))
+                                                        .payingWith(ACCOUNT)
+                                                        .gas(GAS_TO_OFFER)
+                                                        .hasKnownStatus(CONTRACT_REVERT_EXECUTED)
+                                                        .via("UnfreezeTx"),
+                                                cryptoUpdate(ACCOUNT).key(FREEZE_KEY),
```

Comment:
Since the token itself does not have a `FREEZE_KEY`, probably it's obsolete to update the `ACCOUNT` with such a key?